File-Formats/DocIO/Working-with-Table-Of-Contents.md

```diff
@@ -581,10 +581,356 @@ document.Close();
+//Enable a flag to show page numbers in TOC entries.
+tableOfContent.IncludePageNumbers = true;
+//Disable a flag to align page numbers after to the TOC entries.
+tableOfContent.RightAlignPageNumbers = false;
+//Disable a flag to preserve TOC entries without hyperlinks.
+tableOfContent.UseHyperlinks = false;
+//Add a paragraph into the section.
+paragraph = section.AddParagraph();
+//Append text.
+paragraph.AppendText("First ");
+//Append line break.
+paragraph.AppendBreak(BreakType.LineBreak);
+paragraph.AppendText("Chapter");
+//Enable a flag to include new line characters in TOC entries.
+tableOfContent.IncludeNewLineCharacters = true;
+//Set a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading2);
+//Add a section into the Word document.
+section = document.AddSection();
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Append text.
+paragraph.AppendText("Second ");
+paragraph.AppendText("Chapter");
+//Sets a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading1);
+//Add a section into the Word document.
+section = document.AddSection();
+paragraph = section.AddParagraph();
+paragraph.AppendText("Third ");
+paragraph.AppendText("Chapter");
+//Set a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading2);
+section = document.AddSection();
+paragraph = section.AddParagraph();
+paragraph.AppendText("Fourth ");
+paragraph.AppendText("Chapter");
+//Set a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading3);
+//Add the text into the new paragraph of the section.
+section.AddParagraph().AppendText("AdventureWorks Cycles, the fictitious company on which the AdventureWorks sample databases are based, is a large, multinational manufacturing company.");
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Set outline level for paragraph.
+paragraph.ParagraphFormat.OutlineLevel = OutlineLevel.Level2;
+//Append text.
+paragraph.AppendText("Outline Level Paragraph");
+//Enable a flag to consider outline level paragraphs in TOC entries.
+tableOfContent.UseOutlineLevels = true;
+//Add a section into the Word document.
+section = document.AddSection();
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Append a field to the paragraph.
+paragraph.AppendField("Table of Entry Field", FieldType.FieldTOCEntry);
+//Enable a flag to use table entry fields in TOC entries.
+tableOfContent.UseTableEntryFields = true;
+//Update the table of content.
+document.UpdateTableOfContents();
+//Save a Word document to MemoryStream.
+MemoryStream outputStream = new MemoryStream();
+document.Save(outputStream, FormatType.Markdown);
+//Save a stream as a file in the device and invoke it for viewing.
+Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Sample.docx", "application/msword", outputStream);
+}
+
+//Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
```

Comment:
Please download the helper files from the following link to save the stream as a file and open the file for viewing on Xamarin platform.

Keep full stop at the end.